vimiv/imutils/exif.py

```diff
@@ -10,139 +10,269 @@
+
+        except FileNotFoundError:
+            _logger.debug("File %s not found", filename)
             return
 
-        for ifd in self._exif:
-            if ifd == "thumbnail":
-                continue
-
-            for tag in self._exif[ifd]:
-                keyname = piexif.TAGS[ifd][tag]["name"]
-                keytype = piexif.TAGS[ifd][tag]["type"]
-                val = self._exif[ifd][tag]
-                _logger.debug(
-                    f"name: {keyname} type: {keytype} value: {val} tag: {tag}"
-                )
-                if keyname.lower() not in desired_keys:
-                    _logger.debug(f"Ignoring key {keyname}")
+    def get_formatted_exif(self) -> Dict[str, str]:
+
+        desired_keys = [
+            e.strip() for e in api.settings.metadata.current_keyset.value.split(",")
+        ]
+        _logger.debug(f"Read metadata.current_keys {desired_keys}")
+
+        exif = dict()
+
+        for key in desired_keys:
+            # For backwards compability, ssume it has one of the following prefixes
+            for prefix in ["", "Exif.Image.", "Exif.Photo."]:
+                key = f"{prefix}{key}"
+                try:
+                    key_name = self._metadata[key].name
+
+                    try:
+                        key_value = self._metadata[key].human_value
+
+                    # Not all metadata(iptc) provide human_value, take raw_value instead
+                    except AttributeError:
+                        key_value = self._metadata[key].raw_value
+
+                    exif[key] = (key_name, key_value)
+                    break
+
+                except KeyError:
+                    _logger.debug("Key %s is invalid for the current image", key)
                     continue
-                if keytype in (
-                    piexif.TYPES.Byte,
-                    piexif.TYPES.Short,
-                    piexif.TYPES.Long,
-                    piexif.TYPES.SByte,
-                    piexif.TYPES.SShort,
-                    piexif.TYPES.SLong,
-                    piexif.TYPES.Float,
-                    piexif.TYPES.DFloat,
-                ):  # integer and float
-                    self[keyname] = val
-                elif keytype in (
-                    piexif.TYPES.Ascii,
-                    piexif.TYPES.Undefined,
-                ):  # byte encoded
-                    self[keyname] = val.decode()
-                elif keytype in (
-                    piexif.TYPES.Rational,
-                    piexif.TYPES.SRational,
-                ):  # (int, int) <=> numerator, denominator
-                    self[keyname] = f"{val[0]}/{val[1]}"
+
+        return exif
+
+    def copy_exif(self, dest: str, reset_orientation: bool = True) -> None:
+
+        if reset_orientation:
+            with contextlib.suppress(KeyError):
+                self._metadata["Exif.Image.Orientation"] = ExifOrientation.Normal
+        try:
+            self._metadata.copy(dest)
```

Comment:
This crashes when writing. I assume that the py3exiv2 method requires `dest` to be an instance of `pyexiv2.metadata.ImageMetadata` and not simply `str` like we had for piexif.

Reply:
That is true. I should have read the docs more closely. Thanks a lot.

Fixed in 1ea3f98df288fc23ec1507933c96bef65f36830b